### PR TITLE
Don't report redundant feerates to subdaemons

### PR DESCRIPTION
**Pull request by instagibbs**
Currently channeld will get announcements of feerates even if the prior *modified* feerate was the one already reported prior. F.e., the floor is 253, but grabbed rate from bitcoind was 250, but we report it again based on 250(which is never stored!), not 253.

Only report if historical data didn't exist(which is backfilled for smoothing), or if the final computed feerate doesn't match historical.

Changelog-None

### CODE REVIEW

File: lightningd/chaintopology.c [hunk 1]
```diff
@@ -353,16 +353,14 @@ static void update_feerates(struct bitcoind *bitcoind,
 
 		/* Initial smoothed feerate is the polled feerate */
 		if (!old_feerates[i]) {
-			feerate_changed = true;
+            notify_feerate_changed = true;
```
Comment (vincenzopalazzo) [inline lightningd/chaintopology.c:356]:
```suggestion
                       notify_feerate_changed = true;
```

**Review by vincenzopalazzo — COMMENTED** [review]
Just a TAB missing for me

**Review by cdecker — APPROVED** [review]
ACK 3903be15969a81bd3969b85e09bef89c3051149f

**Comment by cdecker** [comment]
Rebased on top of `master` in the hopes that the `mako` issue in lnprototest is fixed.

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/5582/commits/7a518e2a950587d864239e97b42ee5e4cfd54356

**Pull request merged**